vtkKNNGraphFilter.cxx

```diff
@@ -52,9 +52,9 @@ int vtkKNNGraphFilter::RequestData(vtkInformation *vtkNotUsed(request),
   vtkInformation *outInfo = outputVector->GetInformationObject(0);
 
   // get the input and ouptut
-  vtkPointSet* input = vtkPointSet::SafeDownCast(
```

Comment:
vtkPointSet is very different from vtkPoints. vtkPoints is used to store the actual point coordinates inside of things like a vtkPolyData. vtkPointSet, however, is a more abstract container:

http://www.vtk.org/doc/nightly/html/classvtkPointSet.html

I think the reason this was used was to allow normals to be computed on vtkPolyData objects as well as things like vtkStructuredGrid objects.